components/Alert/react/Alert.tsx

```diff
@@ -107,37 +107,34 @@ export const Alert: React.FC<
       : undefined)
 
   const [dismissed, setDismissed] = React.useState(false)
-  const [durationTimeout, setDurationTimeout] = React.useState<
-    number | undefined
-  >(undefined)
+  const durationTimeout = React.useRef<number | undefined>(undefined)
 
-  function clearDurationTimeout() {
+  const clearDurationTimeout = React.useCallback(() => {
     if (durationTimeout) {
-      clearTimeout(durationTimeout)
-      setDurationTimeout(undefined)
+      clearTimeout(durationTimeout.current)
+      durationTimeout.current = undefined
     }
-  }
+  }, [])
 
-  function dismiss() {
+  const dismiss = React.useCallback(() => {
     setDismissed(true)
     onDismiss && onDismiss()
     clearDurationTimeout()
-  }
+  }, [clearDurationTimeout, setDismissed, onDismiss])
 
   React.useEffect(() => {
-    if (onDismiss && duration && !durationTimeout) {
-      setDismissed(false)
+    if (duration && !durationTimeout.current) {
```

Comment:
`duration` is always truthy, since the output of useRef is `{current:undefined}`